app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  MAX_USERNAME_LENGTH = 50
+  MIN_USERNAME_LENGTH = 3
+
+  MIN_PASSWORD_LENGTH = 8
+  PASSWORD_FORMAT = /[a-zA-Z]*/.freeze
+
+  validates :username, allow_blank: false, length: { minimum: MIN_USERNAME_LENGTH, maximum: MAX_USERNAME_LENGTH }
+  validates :password, allow_blank: false, length: { minimum: MIN_PASSWORD_LENGTH },
+                       format: { with: PASSWORD_FORMAT,
+                                 message: 'Password length should be 8 characters, alphanumeric.' }
+
+  def valid_password?(confirmation_password)
+    password.match(confirmation_password)
+  end
```

Comment:
You don't need to verify the match of passwords in API. It's a front-end responsibility. You even don't need the confirmation password parameter

Reply:
i user it to check user password that came from front to back for authentication.